week0/pyomo_introduction.ipynb

```diff
@@ -0,0 +1,1074 @@
+    "You can loop through dictionaries in different ways. For example,\n",
+    "```python\n",
+    "d = {'A': 2, 'B': 4, 'D': 16}\n",
+    "for k in d.keys():\n",
+    "    # loop through the keys in the dictionary\n",
+    "    # access the value with d[k]\n",
+    "    print('key=', k, 'value=', d[k])\n",
+    "    \n",
+    "for v in d.values():\n",
+    "    # loop through the values in the dictionary, ignoring the keys\n",
+    "    print('value=', v)\n",
+    "    \n",
+    "for k,v in d.items():\n",
+    "    # loop through the entries in the dictionary, retrieving both\n",
+    "    # the key and the value\n",
+    "    print('key=', k, 'value=', v)\n",
+    "```\n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Inline Exercise:</b>\n",
+    "The areas listed above for the two states are in square kilometers. Modify the loop below to create a new dictionary that contains the areas in square miles. Print the new dictionary to verify the correct behavior. Note that 1 kilometer is equal to 0.62137 miles.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "tags": [
+     "solution"
+    ]
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'South Dakota': 77120.5214053598, 'Oklahoma': 69897.7360425915}\n"
+     ]
+    }
+   ],
+   "source": [
+    "areas_mi = dict()\n",
+    "for state_name, area in areas.items():\n",
+    "    # Todo: convert the area to sq. mi and assign to the areas_mi dict.\n",
+    "    areas_mi[state_name] = area * (0.62137**2)\n",
+    "print(areas_mi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Python also supports dictionary comprehensions much like list comprehensions. For example:\n",
+    "```python\n",
+    "d = {'A': 2, 'B': 4, 'D': 16}\n",
+    "d2 = {k:v**2 for k,v in d.items()}\n",
+    "```\n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Inline Exercise:</b>\n",
+    "Redo the conversion from square kilometers to square miles using a dictionary comprehension.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "tags": [
+     "solution"
+    ]
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'South Dakota': 77120.5214053598, 'Oklahoma': 69897.7360425915}\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Todo: define areas_mi using a dictionary comprehension and print the result\n",
+    "areas_mi = {k: v * (0.62137**2) for k, v in areas.items()}\n",
+    "print(areas_mi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Matplotlib for generating figures\n",
+    "\n",
+    "We will now briefly explore the use of the `matplotlib` package to generate figures. Before we do this, we will introduce some other helpful tools.\n",
```

Comment:
Not sure if it's necessary to cover Matplotlib specifically.

Reply:
I don't think it hurts? It's relatively simple.